File: src/main/java/textGame/Npc.java
package textGame;

public class Npc extends Character implements Attack, Searchable, Lootable {
    private int damage;
    private Inventory inventory = new Inventory();

    public void setDamage(int damage) {
        this.damage = damage;
    }
    @Override
    public void onDeath() {
        if (isAlive) {
            System.out.println(getName() + " has died.");
            setName(getName() + " (dead)");
            isAlive = false;
        }
        else {
            System.out.println(getName() + " is already dead.");
        }
    }
    @Override
    public int getDamage() {
        return damage;
    }

    @Override
    public String toString() {
        return getName();
    }
    public void takeTurn() {
        attack(getScene().getPlayer());
    }

    @Override
    public Inventory getInventory() {
        return inventory;
    }
}

File: src/main/java/textGame/Npc.kt
package textGame

class Npc : Character(), Attack, Searchable, Lootable {
    private var damage = 0
    private val inventory = Inventory(3)

    fun setDamage(damage: Int) {
        this.damage = damage
    }

    override fun onDeath() {
        if (isAlive) {
            println(getName() + " has died.")
            setName(getName() + " (dead)")
            isAlive = false
        } else {
            println(getName() + " is already dead.")
        }
    }

    override fun getDamage(): Int {
        return damage
    }

    override fun toString(): String {
        return getName()
    }

    fun takeTurn() {
        attack(scene.player)
    }

    override fun getInventory(): Inventory {
        return inventory
    }
}